extensions/resteasy/deployment/src/main/java/io/quarkus/resteasy/deployment/ResteasyScanningProcessor.java

```diff
@@ -344,6 +357,45 @@ public void build(
             if (appClass != null) {
                 servletContextParams.produce(new ServletInitParamBuildItem(JAX_RS_APPLICATION_PARAMETER_NAME, appClass));
             }
+
+            // generate default constructors for suitable concrete @Path classes that don't have them
+            // see https://issues.jboss.org/browse/RESTEASY-2183
+            for (ClassInfo classInfo : withoutDefaultCtor) {
+                // keep it super simple - only generate default constructor is the object is a direct descendant of Object
+                if (!(classInfo.superClassType() != null && classInfo.superClassType().name().equals(DotNames.OBJECT))) {
+                    return;
+                }
+                // again keep it very very simple, we know that there is the @Path annotation, so don't try to generate the
+                // default constructor if there are others (basically we want to avoid surprises with any CDI scopes)
+                if (classInfo.classAnnotations().size() > 1) {
```

Comment:
Hm, so it will stop working if I add e.g. `@Consumes("application/json")` which is a very common use case.

Reply:
Good catch! What other common cases can you think of that we should take into account?

Reply:
Any jax-rs annotation?

Reply:
Makes sense!

Reply:
Fixed